src/Microsoft.ServiceFabric.AspNetCore/AspNetCoreCommunicationListener.cs

```diff
@@ -77,10 +114,15 @@ public string UrlSuffix
         /// </summary>
         public virtual void Abort()
         {
-            if (this.webHost != null)
+            if (this.hostType == HostType.WebHost && this.webHost != null)
             {
                 this.webHost.Dispose();
             }
+            else if (this.hostType == HostType.GenericHost && this.host != null)
+            {
+                this.host.StopAsync().GetAwaiter().GetResult();
```

Comment:
Like webhost dispose in Abort(), we can dispose generic host as well instead of attempting a graceful stop.

Reply:
Removed